lib/evervault/http/request.rb

```diff
@@ -25,32 +25,51 @@ def delete(path, params)
         execute(:delete, build_url(path), params)
       end
 
-      def post(path, params, cage_run: false)
-        execute(:post, build_url(path, cage_run), params)
+      def post(path, params, options: {}, cage_run: false)
+        optional_headers = build_cage_run_headers(options, cage_run)
+        execute(:post, build_url(path, cage_run), params, optional_headers)
       end
 
       private def build_url(path, cage_run = false)
         return "#{@base_url}#{path}" unless cage_run
         "#{@cage_run_url}#{path}"
       end
 
-      def execute(method, url, params)
+      def execute(method, url, params, optional_headers = {})
         resp = Faraday.send(method, url) do |req|
             req.body = params.nil? || params.empty? ? nil : params.to_json
-            req.headers = build_headers
+            req.headers = build_headers(optional_headers)
         end
         return JSON.parse(resp.body) if resp.status >= 200 && resp.status <= 300
         Evervault::Errors::ErrorMap.raise_errors_on_failure(resp.status, resp.body)
       end
 
-      private def build_headers
-        {
+      private def build_headers(optional_headers = {})
```

Comment:
```suggestion
      private def build_headers(optional_headers)
```

We don't need this extra default now realisitcally seeing as we're setting it elsewhere